engines/bastion_katello/app/assets/javascripts/bastion_katello/capsule-content/capsule-content.controller.js

```diff
@@ -0,0 +1,167 @@
+/**
```

Comment:
Something to consider, there is a lot of code in here that is 'logic' code that lives outside of the need to dictate what the view is doing and would be better off inside a service. You could think in some ways of a service as similar to a Rails helper, or a service object -- it holds business logic (ideally re-usable but we don't always know what we will need to re-use) that is not tied to a specific controller allowing the controller to be thin.